components/MediList/index.js

```diff
@@ -2,52 +2,93 @@ import styled from 'styled-components';
 import { useState } from 'react';
 import { dummyMedications } from '../../dummydata';
 
-
 export default function MediList() {
-  const [medikament, setMedikament] = useState('');
-  const [dosage, setDosage] = useState('');
-
-  const handleAddMedikament = (e) => {
-    e.preventDefault();
-    const newMedikament = {
-      id: dummyMedications.length + 1,
-      medicationName: medikament,
-      dosage: dosage,
-    };
-    setMedikament('');
-    setDosage('');
+  const [medications, setMedications] = useState(dummyMedications);
+  const [medication, setMedication] = useState({ medicationName: '', dosage: '' });
```

Comment:
Maybe you could find a better naming for one of these because they are very similar and your code gets hard to read